hiss/team/views.py

```diff
@@ -68,14 +84,25 @@ def form_valid(self, form: JoinTeamForm):
         return super().form_valid(form)
 
 
-class DetailTeamView(shared_mixins.LoginRequiredAndAppliedMixin, generic.DetailView):
+class DetailTeamView(shared_mixins.UserHasTeamMixin, generic.DetailView):
     """
     Renders a Team if the User is a member.
     """
 
     model = Team
     template_name = "team/team_detail.html"
 
+    def handle_no_permission(self):
+        if not self.request.user.is_authenticated:
+            return redirect(
+                f"{reverse_lazy('customauth:login')}?next={reverse_lazy('team:detail', args=[uuid.UUID(self.request.path_info[6:])])}"
```

Comment:
This string-slicing is fragile. A more flexible, Django-way would be [to use the `kwargs` field](https://stackoverflow.com/a/13528732):

```suggestion
                f"{reverse_lazy('customauth:login')}?next={reverse_lazy('team:detail', args=[self.kwargs["pk"])}"
```